ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -160,29 +187,92 @@ internal class CustomerCenterViewModelImpl(
         }
     }
 
-    private suspend fun loadPurchaseInformation(): PurchaseInformation? {
+    private suspend fun loadPurchaseInformation(
+        dateFormatter: DateFormatter,
+        locale: Locale,
+    ): PurchaseInformation? {
         val customerInfo = purchases.awaitCustomerInfo(fetchPolicy = CacheFetchPolicy.FETCH_CURRENT)
 
-        // Customer Center WIP: update when we have subscription information in CustomerInfo
-        val activeEntitlement = customerInfo.entitlements.active.values.firstOrNull()
-        if (activeEntitlement != null) {
-            val product = purchases.awaitGetProduct(activeEntitlement.productIdentifier).first()
-            val locale = getDefaultLocales().first()
-            val purchaseInformation = PurchaseInformation(
-                title = product.description,
-                durationTitle = product.period?.localizedPeriod(locale) ?: "",
-                price = product.price.formatted,
-                expirationDateString = activeEntitlement.expirationDate.toString(),
-                willRenew = activeEntitlement.willRenew,
-                active = activeEntitlement.isActive,
-                productId = activeEntitlement.productIdentifier,
-            )
-            return purchaseInformation
+        val hasActiveProducts = customerInfo.activeSubscriptions.isNotEmpty() ||
+            customerInfo.nonSubscriptionTransactions.isNotEmpty()
```

Comment:
Hmm I don't know if `active` is the best word for non subscriptions TBH... Since this could potentially have been consumed already... I'm not sure of how to word it however, since we don't actually include expired subscriptions... Maybe `hasNonExpiredPurchases`? Or maybe we should separate the non subscriptions from the subscriptions so the wording here makes more sense?